scripts/validate-requirements.sh

```diff
@@ -0,0 +1,236 @@
+        ERRORS+=$'gcloud end user credential not configured.\n'
+    fi
+
+    APPLICATION_DEFAULT_CREDENTIAL_OUTPUT="$(gcloud auth application-default print-access-token 2>&1 >/dev/null)"
+    if [ "$(echo "$APPLICATION_DEFAULT_CREDENTIAL_OUTPUT" | grep -c 'Could not automatically determine credentials')" -eq 1 ]; then
+        echo "You must configure an Application Default Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud application default credential not configured.\n'
+    fi
+}
+
+# Function to validate the Configuration of the Gcloud CLI
+function validate_credential_roles(){
+    check_org_level_roles "$END_USER_CREDENTIAL" "$ORGANIZATION_ID"
+    check_billing_account_roles "$END_USER_CREDENTIAL" "$BILLING_ACCOUNT"
+}
+
+# Verifies wheter a user is assigned to the expected Orgzanization Level roles
+function check_org_level_roles(){
+
+    ORG_LEVEL_ROLES_OUTPUT=$(
+        gcloud organizations get-iam-policy "$2" \
+        --filter="bindings.members:$1" \
+        --flatten="bindings[].members" \
+        --format="table(bindings.role)" 2>/dev/null)
+
+    lines=$(echo "$ORG_LEVEL_ROLES_OUTPUT" | grep -o -e roles/resourcemanager.folderCreator -e roles/resourcemanager.organizationAdmin | wc -l)
+
+    if [ "$lines" -ne 2 ]; then
+        echo "The User must have the Organization Roles resourcemanager.folderCreator and resourcemanager.organizationAdmin"
+        ERRORS+=$'There are missing organization level roles on the Credential.\n'
+    fi
+}
+
+# Verifies wheter a user is assigned to the expected Billing Level roles
+function check_billing_account_roles(){
+
+    BILLING_LEVEL_ROLES_OUTPUT=$(
+        gcloud beta billing accounts get-iam-policy "$2" \
+        --filter="bindings.members:$1" \
+        --flatten="bindings[].members" \
+        --format="table(bindings.role)" 2>/dev/null)
+
+    lines=$(echo "$BILLING_LEVEL_ROLES_OUTPUT" | grep -o -e roles/billing.admin | wc -l)
+
+    if [ "$lines" -ne 1 ]; then
+        echo "The User must have the Billing Account Role billing.admin"
+        ERRORS+=$'There are missing billing account level roles on the Credential.\n'
+    fi
+}
+
+# Checks if initial config was done for 0-bootstrap step
+function validate_bootstrap_step(){
+    FILE=0-bootstrap/terraform.tfvars
+    if [ ! -f "$FILE" ]; then
+        echo "$FILE does not exists."
+    else
+        if [ "$(grep -c REPLACE_ME $FILE)" != 0 ]; then
+            echo "$FILE must have required values fullfiled."
+            ERRORS+=$'terraform.tfvars file must be correctly fullfiled for 0-bootstrap step.\n'
+        fi
+    fi
+}
+
+function main(){
+
+    if [ -z "$END_USER_CREDENTIAL" ] || [ -z "$ORGANIZATION_ID" ] || [ -z "$BILLING_ACCOUNT" ]; then
```

Comment:
it looks better if you create and use a `usage` function like in
https://github.com/terraform-google-modules/terraform-google-project-factory/blob/616ede9456cc8f86ef7995192af3473d17ee7946/helpers/setup-sa.sh#L26

```
usage() {
  echo
  echo " Usage:"
  echo "     $0 -o <organization id> -p <project id> [-b <billing account id>] [-f <folder id>] [-n <service account name>]"
  echo "         organization id        (required)"
  echo "         project id             (required)"
  echo "         billing account id     (optional)"
  echo "         folder id              (optional)"
  echo "         service account name   (optional)"
  echo
  echo " The billing account id is required if owned by a different organization"
  echo " than the Seed Project organization."
  echo
  exit 1
}
```

Reply:
Awesome idea, I haven't think about something like this